packages/cli/test/cases/build.data.graph/src/pages/blog/first-post.md

```diff
@@ -1,5 +1,7 @@
 ---
 template: 'blog'
+title: 'First'
```

Comment:
Any reason we need to change / add the title here?  Or if so, can we change it to something a little more descriptive like  _First Post_, _Second Post_, etc?

Reply:
It must have a title for the shelf/header/anymenu if its being used in a menu.

Reply:
Ah, for some reason I thought title was "inferred" / generated by the graph lifecycle, but that is [not the case](https://github.com/ProjectEvergreen/greenwood/blob/release/0.5.0/packages/cli/src/lifecycles/graph.js#L75) as I had thought.  

Maybe it could / should be?  (can make a new issue for that)

Reply:
I see, I did it as [part of the graph](https://github.com/ProjectEvergreen/greenwood/blob/release/0.5.0/packages/cli/src/data/schema/graph.js#L23).  I think it would make more sense to do this within the graph lifecycle instead.  I think it's best to try and keep our resolvers as simple as possible and better not split up this kind of logic into multiples places.

Will definitely make an issue for this, should be pretty simple and quick and can be done post release.

Reply:
While yes it saves a step when we infer the title from a heading, it's also not necessarily what a person wants as the anchor text in their menu. E.g.

```
---
title: "Interesting Stuff"
---
# Interesting Stuff about something you want to know
```

Reply:
Yes, understood, and that already works fine right now.  But I'm just observing that in _this_ case it would be perfectly sufficient. 

Instead of just setting it to an empty string, we just pull the resolver logic "closer".
```
# currently
const title = title || '';

# proposal
const title = title || label;
```

If a title is set, use it.  Otherwise, fallback to a default.  (that's anything but an empty string).

Reply:
Yes that woud work.